dts/bindings/flash_controller/nxp,k64f-flash-controller.yaml

```diff
@@ -0,0 +1,19 @@
+---
+title: NXP K64 Flash Memory Controller (FMC)
+id: nxp,k64f-flash-controller.yaml
+version: 0.1
+
+description: >
+    This binding gives a base representation of the NXP K64 FMC
+
+inherits:
+    !include flash-controller.yaml
+
+properties:
+    compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "nxp,k64f-flash-controller"
```

Comment:
This flash controller is used across the Kinetis family. Let's change the constraint to `nxp,kinetis-flash-controller` so we don't have k64 references in kw2xd, etc.

Reply:
> This flash controller is used across the Kinetis family. Let's change the constraint to nxp,kinetis-flash-controller so we don't have k64 references in kw2xd, etc.

So looking at the register map the KW40/KW41/KL25 seem to be a bit different from K64/KW2XD.  So wondering if there should be 2 different compatibles, one for k64/kw2xd and one for kw40/kw41/kl25.

Reply:
You're right. The mcux flash driver abstracts the hardware differences so I mistakenly thought the flash controller was the same. Looking more closely at the register maps and hardware reference manuals, there should be three different compatibles:
nxp,kinetis-ftfa (kl25, kw40, kw41)
nxp,kinetis-ftfe (k64)
nxp,kinetis-ftfl (kw2xd)

Reply:
Doe the 'a' 'e' 'l' have any meaning?

Reply:
The 'a' 'e' 'l' variants are selected based on the series (Kinetis L-series vs. K-series) and the size of the flash (greater than or less than 512 KB). I haven't been able to track down an explanation of why those particular letters or if they are an abbreviation for something.
https://www.nxp.com/docs/en/application-note/AN4835.pdf